feat(components): added checkbox component

davidruvolo51 opened the pull request:
What are the main changes you did:
- Added checkbox component using tailwind styles
- Revised `InputLabel` component: simplified parameters and moved label to slot
- Fixed required styling
- Added stories for checkbox component and input label

how to test:
- explain here what to do to test this (or point to unit tests)

todo:
- [ ] Check for uses of input label in the nuxt-ssr catalogue
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

mswertz commented on apps/nuxt3-ssr/components/ContactForm.vue:
```diff
@@ -14,11 +14,9 @@ defineEmits(["submitForm"]);
-        :label="field.label"
-      />
+      <InputLabel class="pl-3 text-body-base" :for="field.name">
+        {{ field.label }}
```
so you want to keep door open for html rich labels?

mswertz replied:
ok until proven otherwise :-)

mswertz commented on apps/nuxt3-ssr/components/ContactForm.vue:
```diff
@@ -38,7 +36,7 @@ defineEmits(["submitForm"]);
         :options="(field as ISelectFormField).options"
       />
-      <div class="pl-3" :class="{ 'text-red-500': field.hasError }">
+      <div class="pl-3" :class="{ 'text-invalid': field.hasError }">
```
I like semantic styles. Should be a guideiline for our lib

davidruvolo51 replied:
Yes, that's a good idea. In the next PR, I'll start writing this out in the README

mswertz commented on apps/tailwind-components/components/input/Label.vue:
```diff
@@ -21,15 +18,15 @@ withDefaults(
 
 <template>
   <label
-    :for="for"
```
we need the 'for' for the ARIA right?

mswertz replied:
do I understand it doesn't need a prop (which makes sense but might be better to have it as prop to help us to comply?)

davidruvolo51 replied:
It's for ensuring we follow good ARIA and semantic HTML practices. I put it as a required prop so that typescript will throw an error if this is missing.

mswertz replied:
great

mswertz commented on apps/tailwind-components/pages/input/Label.story.vue:
```diff
@@ -0,0 +1,57 @@
+<template>
+  <h2>Label component</h2>
+  <p>
+    The <code>InputLabel</code>component is a wrapper around the
```
lovely docs

mswertz commented on apps/tailwind-components/pages/input/Label.story.vue:
```diff
@@ -0,0 +1,57 @@
+  >
+    <div>
+      <h3>Input Label: default use</h3>
+      <InputLabel for="input-string-default">
```
right

mswertz reviewed (commented):
Nice. Small comments inline.

I would like a way to have a label and a description/explanation, and where error messages go. So I can understand the anatomy of the inputs in the their full glory. Also I would recommend that we make a standard wrapper for our inputs therefore?
(I can imagine you want to have a 'smart' component therefore that allows you to do all this on one go). 

Also I notice that the h3 titles in the stories distract me (I first thought those were the label). So in the way components are demoed I propose we should put the title outside the demo panel somehow.

I.e. now this thing seems confusing but the h3 looks like it is the label and the actual label lightly rendered as description
<img width="466" alt="image" src="https://github.com/user-attachments/assets/f80a8f44-ec9b-4b08-95c5-49bac77823f2">

mswertz reviewed (changes requested):
I think you still need to add the new components to the menu?

I don't understand why we don't generate the menu from the list of stories, considering the folders, but I suppose for now you would need to put them here?
https://github.com/molgenis/molgenis-emx2/blob/feat/tw-checkbox/apps/tailwind-components/app.vue

davidruvolo51 commented:
For autogenerating the component lists, you can use this:

```js
// in the setup
const modules = import.meta.glob("./**/*.story.vue", {
  import: "default",
  eager: true,
});
const stories = Object.keys(modules).map((module) => {
  const name: string = module.split('/').reverse()[0];
  const path: string = module.replace('./pages/', '/');
  return {
    name: name.replace('.story.vue',''),
    dir: path.split('/').filter((path: string) => path !== "" && path !== name)[0],
    path: path.replace('.vue',''),
  }
});
```

And then in the template -

```html
<ul class="list-none">
    <li class="py-2">
        <NuxtLink class="hover:underline" to="/">Home</NuxtLink>
    </li>
</ul>
```
